replace colnames with names in tidy.polr (#497)

haozhu233 opened the pull request:
Trying to fix #497

alexpghayes commented:
Thanks for the PR! Would you mind adding a quick test and adding the bugfix to `NEWS` and then I'll merge?

haozhu233 commented:
@alexpghayes I just added NEWs and test cases and I made some changes to how I fixed this. The main problem is that when there is only one numeric x, `stats::confint()` returns a vector instead of a matrix.

alexpghayes commented:
Thanks for taking this on!
